src/core/path.c

```diff
@@ -342,12 +342,18 @@ static int path_add_default_dependencies(Path *p) {
                 return r;
 
         if (MANAGER_IS_SYSTEM(UNIT(p)->manager)) {
-                r = unit_add_two_dependencies_by_name(UNIT(p), UNIT_AFTER, UNIT_REQUIRES, SPECIAL_SYSINIT_TARGET, true, UNIT_DEPENDENCY_DEFAULT);
+                r = unit_add_two_dependencies_by_name(
+                                UNIT(p),
+                                UNIT_AFTER,
+                                UNIT(p)->survive_system_transition ? _UNIT_DEPENDENCY_INVALID : UNIT_REQUIRES,
+                                SPECIAL_SYSINIT_TARGET,
+                                /* add_reference= */ true,
+                                UNIT_DEPENDENCY_DEFAULT);
                 if (r < 0)
                         return r;
         }
```

Comment:
I'd like to split out this pattern as a new helper function.

Reply:
The issue is that it's not the same everywhere, it changes depending on the unit type